internal/storage/storageutil/custom_retry.go

```diff
@@ -0,0 +1,78 @@
+package storageutil
+
+import (
+	"io"
+	"net"
+	"net/url"
+
+	"google.golang.org/api/googleapi"
+)
+
+func ShouldRetry(err error) (b bool) {
+	// HTTP 50x errors.
+	if typed, ok := err.(*googleapi.Error); ok {
+		if typed.Code >= 500 && typed.Code < 600 {
+			b = true
+			return
+		}
+	}
+
+	// HTTP 429 errors (GCS uses these for rate limiting).
+	if typed, ok := err.(*googleapi.Error); ok {
+		if typed.Code == 429 {
+			b = true
+			return
+		}
+	}
+
+	// HTTP 401 errors - Invalid Credentials
+	// This is a work-around to fix the rare issue, we get while executing the
+	// long ml-based test on gcsfuse. The better fix would be to expire the
```

Comment:
this issue might happen for non ml-based as well if gcsfuse is running for a long time and never stopped. Just mention that this is a workaround to fix the corner case where <mention method which returns token as valid with code link> returns the token as valid but GCS says invalid. This might happen because of client-server timer issues.

Actual fix will be refresh the token earlier than 1hr. Changes will be done post resolution of this issue: